tests/test_views.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+from django.test import RequestFactory
+
+from rest_framework_json_api import serializers, views
+from rest_framework_json_api.relations import ResourceRelatedField
+from rest_framework_json_api.utils import format_value
+
+from .models import BasicModel
+
+related_model_field_name = "related_field_model"
+
+
+@pytest.mark.urls("tests.urls")
+@pytest.mark.parametrize(
+    "format_links",
+    [
+        None,
+        "dasherize",
+        "camelize",
+        "capitalize",
+        "underscore",
+    ],
+)
+def test_get_related_field_name_handles_formatted_link_segments(format_links):
+    url_segment = format_value(related_model_field_name, format_links)
+
+    request = RequestFactory().get("/basic_models/1/{}".format(url_segment))
```

Comment:
use [RequestFactory fixture](https://pytest-django.readthedocs.io/en/latest/helpers.html#rf-requestfactory) of pytest-django so it is extendible if we need to in the future. Also a good stop to use [f-strings](https://realpython.com/python-f-strings/#simple-syntax)